### ISSUE TITLE
Improve validation of command-line options

### ISSUE THREAD

**Issue by mjordan** [MarcusBarnes/mik#203]
Currently, there is little feedback from MIK if the user misspells an .ini file, or worse, misspells 'limit' (in which case MIK starts up and doesn't apply a limit).

We could perform better option validation, or incorporate a CLI library like https://github.com/nategood/commando/ that does this automatically. We'd want a CLI library that didn't interfere with our class loading or other code-level logic. We should check out the Symfony Console Component but I wouldn't want us to be forced to adopt the entire Symfony ecosystem. It's awesome but overkill for MIK.

**Comment by MarcusBarnes** [comment]
:thumbsup:

**Comment by mjordan** [comment]
I've integrated the Commando component (linked above) into `mik` and it appears to work as intended, on Linux at least (haven't tested yet on Windows). So, now, `./mik  --help` produces:

```
 ./mik                                                                                                                    


--config/-c <argument>
     Required. Path to MIK configuration file.


--checkconfig/--cc <argument>
     Applies checks to configuration, does not process objects. Must be one of 'snippets', 'urls', 'paths', 'aliases',
     'input_directories', or 'all'.


--help
     Show the help page for this command.


--limit/-l <argument>
     Number of objects to process.
```

We get aliases, so that `./mik  --c tutorial_config.ini` and `./mik  --config tutorial_config.ini` are equivalent.

And, most importantly, we get option validation, so that `./mik  --config tutorial_config.ini --limil 10`  produces:

```
ERROR: Unknown option, limil, specified
```
instead of running without a limit.

Only drawback I can see is that Commando doesn't support the `=` in options, which means that MIK's basic command syntax changes. However, I would argue that using Commando results in a safer, much more conventional command-line user interface than what we had before. But, this change can be documented in the README.

**Comment by mjordan** [comment]
To test, switch to the issue-203 branch, then run `composer update`, since we are adding

```
"require": {
    "nategood/commando": "*"
}
```

to composer.json.

**Comment by mjordan** [comment]
AFAICT, only README.md and the tutorial (including screenshots) would need to be updated to reflect the change in command-line syntax.

**Comment by MarcusBarnes** [comment]
@mjordan Thanks.  I will schedule some time to test.

**Comment by mjordan** [comment]
Excellent thanks. Just did a smoke test on Windows and everything appears to work as expected.

**Comment by MarcusBarnes** [comment]
@mjordan Passed my initial basic tests.

**Comment by mjordan** [comment]
As discussed offline, I've tested this pretty thoroughly so I'll merge into master.

**Comment by mjordan** [comment]
Merged with  72b5d88; reopening because updating tutorial is yet to be done.

**Comment by mjordan** [comment]
Tutorial has been updated to show no `=` in MIK commands. Closing.
